jlc_kicad_tools/jlc_lib/cpl_fix_rotations.py

```diff
@@ -66,11 +72,20 @@ def FixRotations(input_filename, output_filename, db):
           if row[i] in HEADER_REPLACEMENT_TABLE:
             row[i] = HEADER_REPLACEMENT_TABLE[row[i]]
       else:
+        rotation = float(row[rotation_index])
+        rotation_changed = False
+        if posx_index is not None and side_index is not None:
+            if row[side_index].strip() == "bottom":
+                row[posx_index] = "{0:.6f}".format(-float(row[posx_index]))
+                rotation = (rotation + 180) % 360
         for pattern, correction in db.items():
           if pattern.match(row[package_index]):
             logging.info("Footprint {} matched {}. Applying {} deg correction"
                 .format(row[package_index], pattern.pattern, correction))
-            row[rotation_index] = "{0:.6f}".format((float(row[rotation_index]) + correction) % 360)
+            rotation = (rotation + correction) % 360
+            rotation_changed = True
             break
+        if rotation_changed:
```

Comment:
Why not always apply the change?

Reply:
I guess this if is not necessary for correct functionality, but:
1) It's more efficient. Doesn't have to run .format
2) It is the same behaviour from before this PR, where row[rotation_index] is modified only if it has to change. Before this PR there was no rotation_changed variable, but it was behaving the same because the row[rotation_index] = assignment was inside the above if statement. I didn't want to change unintended behaviour.

Removing this if would mean that say if rotation is not changed on a component, then the number format might still change.

Reply:
Cool thanks. I would say for simplicity let's just change it all the time. If it reveals a bug, it would have to be fixed anyways (since otherwise components on the back side would be buggy).

Reply:
Oh, I just looked at it again, and there may be a bug here,
looks like I forgot to do 'rotation_changed = True' on bottom components.
I'll investigate and update this PR.